### ISSUE TITLE
Build Failed: Killed kubectl. Hit timeout of 15s.

### ISSUE THREAD

**Issue by zkutasi** [tilt-dev/tilt#3491]
Hi,

Currently I have a very complex deployment consisting of around a 100 kubernetes objects, 12 of them Pod type, rest of them seems to be the "Uncategorized" type. Tilt gives this error in the "Uncategorized" part, which makes me think this many objects is just to much for the API server to handle in 15 seconds. The other idea I had is that maybe one resource takes too much time... but which one? I am having issues actually deploying anything because even the namespace creation got stuck due to I think the kill command once... I try to reload on the GUI hoping that once it will succeed, but this is not ideal.

My questions would be:
Would it be possible to somehow debug what is going on ?
Would it be possible to increase the 15s timeout (or make it configurable)? Maybe I am having way too much stuff to deploy.

EDIT: I have tried out with fewer dependencies to see if something accumulates or not. with ~45 objects, it took 14 seconds (1 out of 2 tries it was timing out). 28 objects, 14 seconds... 19 objects, 10 seconds... 8 objects 3 seconds.

Thanks.

**Comment by zkutasi** [comment]
Never mind it seems... in 0.15.0... the counter goes further than 15 seconds... for ~45 objects, it took 20 seconds, but it was successful at the end. I have tried out with the many-many objects, and I got a result after 23 seconds... seems better now.

**Comment by nicks** [comment]
thanks for the report! Ya, if you hit this again, there's a Tiltfile API to increase the timeout
https://docs.tilt.dev/api.html#api.update_settings

though if you've got 45 objects in the uncategorized bucket, i wonder if we need better APIs to help you group them or need to break them up across multiple kubectl calls cc @jazzdan

**Comment by jazzdan** [comment]
Interesting! @zkutasi just for my own tracking purpose, of those 80 or so uncategorized objects, what types of Kubernetes objects were they? Any CRDs?

Thanks!

**Comment by zkutasi** [comment]
Hi... In it it had a pack of CRDs (5), a bunch of Roles, ServiceAccounts, RoleBindings (30, 10 each), ConfigMaps (8-ish), Secrets (around 10), Endpoints (6).... and a few others. This deployment is a POC that I am doing in the company that houses 4 microservices, 3 PGSQL RDBMS, and an mTLS stack... all proprietary code.

I already started looking into the grouping, there is an API for that it seems... but I could maybe separate these out via types... for example Roles+SAs+RoleBindings could co into one... since all of this is made with helm, I could not easily do it with k8s_resource's ability to re-group, as I do not know the names of the blobs and the grouping is working on tilt-names as I saw it.
